libass/ass.c

```diff
@@ -186,15 +188,18 @@ static int resize_read_order_bitmap(ASS_Track *track, int max_id)
     return -1;
 }
 
-static int test_and_set_read_order_bit(ASS_Track *track, int id)
+static int test_and_write_read_order_bit(ASS_Track *track, int id, bool set)
 {
     if (resize_read_order_bitmap(track, id) < 0)
         return -1;
-    int index = id / 32;
-    uint32_t bit = 1u << (id % 32);
-    if (track->parser_priv->read_order_bitmap[index] & bit)
+    int index = id >> 5; //div32
+    uint32_t bit = 1u << (id & 0x1F); //mod32
+    if (set && track->parser_priv->read_order_bitmap[index] & bit)
         return 1;
-    track->parser_priv->read_order_bitmap[index] |= bit;
+    if (set)
+        track->parser_priv->read_order_bitmap[index] |= bit;
+    else
+        track->parser_priv->read_order_bitmap[index] &= ~bit;
     return 0;
```

Comment:
Optimisation of the division and modulo suggests you view this as perf-critical, but `set` introduces new conditions and branching options. `set` is always passed as a constant though, so if perf is relevant we could split out computaton of `index` and `bit` as a `static inline` function and have two different functions for test-and-set and clearing of a bit.

Reply:
I changed those to bitwise operation solely based on the context of the function: working on bits. Division and module are better candidates for other contexts. No performance considerations were made otherwise. I will split the set and clear functionalities in two separate functions, as it reduces the complexity anyway.